internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state"
+)
+
+type Collector interface {
+	CollectData() Data
+}
+
+type DataCollector struct {
+	ctx       context.Context
+	k8sClient client.Client
+	processor state.ChangeProcessor
+}
+
+type GraphResourceCount struct {
+	Gateways           int
+	GatewayClasses     int
+	HTTPRoutes         int
+	ReferencedSecrets  int
```

Comment:
I wonder if we really need to use Referenced prefix. HTTPRoutes, Gateways, GatewayClass - we don't report total count of those resources, only the ones used by NGF, and we don't use any prefix. I think it is ok to assume that Secrets are ReferencedSecrets and the same for ReferencedServices